src/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs

```diff
@@ -217,8 +220,11 @@ protected override void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _keys?.Dispose();
-                    _keys = null;
+                    if (_keys != null)
+                    {
+                        _keys.Dispose();
```

Comment:
What if this method is called in parallel? This might potentially throw a NRE. The old solution does not.